### ISSUE TITLE
Can't save or load story in KoboldAI

### ISSUE THREAD

**Issue by ElnuDev** [nixified-ai/flake#31]
In KoboldAI, I can't save my story. I'm on NixOS. It always gives a "read-only file system" error due to probably trying to save the file into the Nix store rather than somewhere in the home directory (shouldn't it be going somewhere in `~/.koboldai`?).

I can save the story as JSON, but this is useless because I get the following error in logs when I try to import it in again:

```
Exception in thread Thread-29 (_handle_event_internal):
Traceback (most recent call last):
  File "/nix/store/iw1vmh509hcbby8dbpsaanbri4zsq7dj-python3-3.10.10/lib/python3.10/threading.py", line 1016, in _bootstrap_inner
    self.run()
  File "/nix/store/iw1vmh509hcbby8dbpsaanbri4zsq7dj-python3-3.10.10/lib/python3.10/threading.py", line 953, in run
    self._target(*self._args, **self._kwargs)
  File "/nix/store/blsb0ajywpv3ahbzxwaf2d23r77cxb5n-python3-3.10.10-env/lib/python3.10/site-packages/socketio/server.py", line 731, in _handle_event_internal
    r = server._trigger_event(data[0], namespace, sid, *data[1:])
  File "/nix/store/blsb0ajywpv3ahbzxwaf2d23r77cxb5n-python3-3.10.10-env/lib/python3.10/site-packages/socketio/server.py", line 756, in _trigger_event
    return self.handlers[namespace][event](*args)
  File "/nix/store/blsb0ajywpv3ahbzxwaf2d23r77cxb5n-python3-3.10.10-env/lib/python3.10/site-packages/flask_socketio/__init__.py", line 282, in _handler
    return self._handle_event(handler, message, namespace, sid,
  File "/nix/store/blsb0ajywpv3ahbzxwaf2d23r77cxb5n-python3-3.10.10-env/lib/python3.10/site-packages/flask_socketio/__init__.py", line 766, in _handle_event
    ret = handler(*args)
  File "/nix/store/513r43w6qkrv1ri5m49dawhjvrp205dw-koboldAi-patchedSrc/aiserver.py", line 466, in g
    return f(*a, **k)
  File "/nix/store/513r43w6qkrv1ri5m49dawhjvrp205dw-koboldAi-patchedSrc/aiserver.py", line 3661, in get_message
    loadfromfile()
  File "/nix/store/513r43w6qkrv1ri5m49dawhjvrp205dw-koboldAi-patchedSrc/aiserver.py", line 6496, in loadfromfile
    loadpath = fileops.getloadpath(vars.savedir, "Select Story File", [("Json", "*.json")])
  File "/nix/store/513r43w6qkrv1ri5m49dawhjvrp205dw-koboldAi-patchedSrc/fileops.py", line 32, in getloadpath
    import tkinter as tk
  File "/nix/store/iw1vmh509hcbby8dbpsaanbri4zsq7dj-python3-3.10.10/lib/python3.10/tkinter/__init__.py", line 37, in <module>
    import _tkinter # If this fails your Python may not be configured for Tk
ModuleNotFoundError: No module named '_tkinter'
```

This is really frustrating, because I have to manually copy in all my world data, story, memory, etc. every single time I start up KoboldAI.

Thanks in advance!

**Comment by MatthewCroughan** [comment]
> shouldn't it be going somewhere in ~/.koboldai

Yes, it should be. Take a look at the patches I've made to make that happen for all of the other program state here https://github.com/nixified-ai/flake/blob/master/projects/koboldai/package.nix#L53

I would really appreciate a contribution to fix this, as I haven't got much time anymore to focus on nixified.ai in general. I applied for GitHub sponsors for this project around 3 weeks ago, and they haven't gotten back to me. I'm hoping that better tools that are authored by people who understand how to store/manage state are made so that we can get away from tools like KoboldAI which are badly engineered, so that maintenance like this doesn't need to occur.

Here's an example of a tool that we could switch to and deprecate KoboldAI https://github.com/nixified-ai/flake/pull/29
